src/s.mli

```diff
@@ -0,0 +1,472 @@
+
+  (** Sorted Set commands *)
+
+  (* Add one or more members to a sorted set, or update its score if it already exists. *)
+  val zadd : connection -> string -> (int * string) list -> int IO.t
+
+  (* Return a range of members in a sorted set, by index. *)
+  val zrange : connection -> ?withscores:bool -> string -> int -> int -> reply list IO.t
+
+  (* Return a range of members in a sorted set, by score. *)
+  val zrangebyscore : connection -> ?withscores:bool -> string -> int -> int -> reply list IO.t
+
+  (* Remove one or more members from a sorted set. *)
+  val zrem : connection -> string list -> int IO.t
+
+  (** Transaction commands *)
+
+  (* Marks the start of a transaction block. Subsequent commands will be queued for atomic execution using EXEC. *)
+  val multi : connection -> unit IO.t
+
+  (* Executes all previously queued commands in a transaction and restores the connection state to normal. *)
+  val exec : connection -> reply list IO.t
+
+  (* Flushes all previously queued commands in a transaction and restores the connection state to normal. *)
+  val discard : connection -> unit IO.t
+
+  (* Marks the given keys to be watched for conditional execution of a transaction. *)
+  val watch : connection -> string list -> unit IO.t
+
+  (* Flushes all the previously watched keys for a transaction. *)
+  val unwatch : connection -> unit IO.t
+
+  val queue : (unit -> 'a IO.t) -> unit IO.t
+
+  (** Scripting commands *)
+
+  (* Load the specified Lua script into the script cache. Returns the SHA1 digest of the script for use with EVALSHA. *)
+  val script_load : connection -> string -> string IO.t
+
+  (* Evaluates a script using the built-in Lua interpreter. *)
+  val eval : connection -> string -> string list -> string list -> reply IO.t
+
+  (* Evaluates a script cached on the server side by its SHA1 digest. *)
+  val evalsha : connection -> string -> string list -> string list -> reply IO.t
+
+  (** Server *)
+
+  val bgrewriteaof : connection -> unit IO.t
+
+  val bgsave : connection -> unit IO.t
+
+  val config_resetstat : connection -> unit IO.t
+
+  val dbsize : connection -> int IO.t
+
+  (* clear all databases *)
+  val flushall : connection -> unit IO.t
+
+  (* clear current database *)
+  val flushdb : connection -> unit IO.t
+
+  val info : connection -> (string * string) list IO.t
+
+  (* last successful save as Unix timestamp *)
+  val lastsave : connection -> float IO.t
+
+  (* role in context of replication *)
+  val role : connection -> reply list IO.t
+
+  (* synchronous save *)
+  val save : connection -> unit IO.t
+
+  (* save and shutdown server *)
+  val shutdown : connection -> unit IO.t
+end
+
+module type Cache_params = sig
+  type key
+  type data
+
+  val cache_key : key -> string
+  val cache_expiration : int option
+
+  val data_of_string : string -> data
+  val string_of_data : data -> string
+end
+
+module type Cache = sig
+  module IO : IO
+  module Client : Client
```

Comment:
If you're only using `Client` for `Client.connection`, you should consider just having an abstract `connection` type instead.